trajectory_executor/src/trajectory_executor_node.cpp

```diff
@@ -0,0 +1,185 @@
+
+    // Load parameters
+    get_parameter<double>("trajectory_publish_rate", config_.trajectory_publish_rate);
+    get_parameter<std::string>("default_control_plugin", config_.default_control_plugin);
+    get_parameter<std::string>("default_control_plugin_topic", config_.default_control_plugin_topic);
+
+    RCLCPP_INFO_STREAM(get_logger(), "Loaded params: " << config_);
+
+    // Register runtime parameter update callback
+    add_on_set_parameters_callback(std::bind(&TrajectoryExecutor::parameter_update_callback, this, std_ph::_1));
+
+    // Setup subscribers
+    plan_sub_ = create_subscription<carma_planning_msgs::msg::TrajectoryPlan>("trajectory", 5,
+                                                              std::bind(&TrajectoryExecutor::onNewTrajectoryPlan, this, std_ph::_1));
+    state_sub_ = create_subscription<carma_planning_msgs::msg::GuidanceState>("state", 5,
+                                                              std::bind(&TrajectoryExecutor::guidanceStateMonitor, this, std_ph::_1));
+
+    cur_traj_ = std::unique_ptr<carma_planning_msgs::msg::TrajectoryPlan>();
+
+    RCLCPP_DEBUG_STREAM(get_logger(), "Setting up publishers for control plugin topics...");
+
+    std::map<std::string, carma_ros2_utils::PubPtr<carma_planning_msgs::msg::TrajectoryPlan>> control_plugin_topics;
+    auto discovered_control_plugins = queryControlPlugins();
+
+    for (auto it = discovered_control_plugins.begin(); it != discovered_control_plugins.end(); it++)
+    {
+      RCLCPP_DEBUG_STREAM(get_logger(), "Trajectory executor discovered control plugin " << it->first.c_str() << " listening on topic " <<  it->second.c_str());
+      carma_ros2_utils::PubPtr<carma_planning_msgs::msg::TrajectoryPlan> control_plugin_pub = create_publisher<carma_planning_msgs::msg::TrajectoryPlan>(it->second, 1000);
+      control_plugin_topics.insert(std::make_pair(it->first, control_plugin_pub));
+    }
+
+    traj_publisher_map_ = control_plugin_topics;
+    RCLCPP_DEBUG_STREAM(get_logger(), "TrajectoryExecutor component initialized succesfully!");
+
+    // Return success if everthing initialized successfully
+    return CallbackReturn::SUCCESS;
+  }
+  carma_ros2_utils::CallbackReturn TrajectoryExecutor::handle_on_activate(const rclcpp_lifecycle::State &)
+  {
+    // Create timer for publishing outbound trajectories to the control plugins
+    int timer_period_ms = (1 / config_.trajectory_publish_rate) * 1000; // Conversion from frequency (Hz) to milliseconds time period
+
+    timer_ = create_timer(
+        get_clock(),
+        std::chrono::milliseconds(timer_period_ms), 
+        std::bind(&TrajectoryExecutor::onTrajEmitTick, this));
+
+    return CallbackReturn::SUCCESS;
+  }
+
+  void TrajectoryExecutor::onTrajEmitTick()
+  {
+    std::unique_lock<std::mutex> lock(cur_traj_mutex_);
+    RCLCPP_DEBUG_STREAM(get_logger(), "TrajectoryExecutor tick start!");
+
+    if (cur_traj_ != nullptr) {
+      if (!cur_traj_->trajectory_points.empty()) {
```

Comment:
I guess I missed it during the removal of the trajectory trimming, but this conditional is no longer needed, really. Since we're not shrinking the trajectory itself as we advance along it any more, the only way for us to get an empty trajectory is for that to have been the output from the plan_delegator, which I don't think is possible (might be worth double checking if plan_delegator has safeguards for that).

Reply:
Good catch. I verified that plan_delegator has safeguards in place that restrict it from publishing any trajectory with less than 2 points. I've gone ahead and removed this conditional.